.buildkite/steps/tests.sh

```diff
@@ -10,6 +10,12 @@ package="github.com/buildkite/agent-stack-k8s/v2/internal/integration_test"
 branch="${BUILDKITE_BRANCH:-main}"
 IMAGE=$(buildkite-agent meta-data get agent-image)
 export IMAGE
+# Do NOT allow integration test workloads in the buildkite namespace.
+# It causes confusion and complicates cleanup.
+# The buildkite namespace is reserved for CI workloads.
+# Integration tests will use their own namespace and out-of-cluster k8s controller to manage job pods.
+NAMESPACE=buildkite-k8s-integration-test
+export NAMESPACE
```

Comment:
Not sure why we're not doing this:

```suggestion
export NAMESPACE=buildkite-k8s-integration-test
```

and above:

```diff
-IMAGE=$(buildkite-agent meta-data get agent-image)
-export IMAGE
export IMAGE="$(buildkite-agent meta-data get agent-image)"
```

Reply:
It's a good question. This is one of shellcheck's recommendations.

Please see https://www.shellcheck.net/wiki/SC2155

Reply:
Oh, that makes sense.